scripts/release-tools/downstream/y-stream-setup.sh

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env bash
+set -eoux pipefail
+
+# Check the number of input parameters
+if [ "$#" -ne 2 ]; then
+    echo "Usage: $0 <previous_release> <release>"
+    exit 1
+fi
+
+previous_release="$1"
+release="$2"
+
+# Define the version pattern
+version_pattern="^[0-9]+\.[0-9]+$"
+
+# Check if variables match the version pattern
+if ! [[ $previous_release =~ $version_pattern ]] || ! [[ $release =~ $version_pattern ]]; then
+    echo "Both previous_release and release must match the pattern x.y, where x and y are integers."
+    exit 1
+fi
+
+# Extract major and minor version numbers
+previous_major=$(cut -d '.' -f 1 <<< "$previous_release")
+previous_minor=$(cut -d '.' -f 2 <<< "$previous_release")
+release_major=$(cut -d '.' -f 1 <<< "$release")
+release_minor=$(cut -d '.' -f 2 <<< "$release")
```

Comment:
Remember to always quote the `$()`
```suggestion
# Extract major and minor version numbers
previous_major="$(cut -d '.' -f 1 <<< "$previous_release")"
previous_minor="$(cut -d '.' -f 2 <<< "$previous_release")"
release_major="$(cut -d '.' -f 1 <<< "$release")"
release_minor="$(cut -d '.' -f 2 <<< "$release")"
```

Reply:
I think we have a more advanced python version of such code in the midstream repo. I still think that  stackrox/stackrox is not the best place for this script.